frappe/public/js/frappe/list/list_view.js

```diff
@@ -440,7 +440,7 @@ frappe.views.ListView = class ListView extends frappe.views.BaseList {
 
 		if (this.$result.find('.list-row-head').length === 0) {
 			// append header once
-			this.$result.prepend(this.get_header_html());
+			this.columns && this.$result.prepend(this.get_header_html());
```

Comment:
This should be fixed in `get_header_html`.

Or else you have to fix it in[ `inbox_view.js`](https://github.com/frappe/frappe/blob/develop/frappe/public/js/frappe/views/inbox/inbox_view.js#L80).